src/System.Management.Automation/FormatAndOutput/DefaultFormatters/PowerShellCore_format_ps1xml.cs

```diff
@@ -1625,5 +1637,123 @@ private static IEnumerable<FormatViewDefinition> ViewsOf_Microsoft_PowerShell_Ma
+                                $_.Address
+                            }
+                            else {
+                                '*'
+                            }
+                        ")
+                        .AddScriptBlockColumn(@"
+                            if ($_.Status -eq 'TimedOut') {
+                                '*'
+                            }
+                            else {
+                                $_.Latency
+                            }
+                        ")
+                        .AddScriptBlockColumn(@"
+                            if ($_.Status -eq 'TimedOut') {
+                                '*'
+                            }
+                            else {
+                                $_.BufferSize
+                            }
+                        ")
+                        .AddPropertyColumn("Status")
+                    .EndRowDefinition()
+                    .GroupByProperty("Destination")
+                .EndTable());
+        }
+
+        private static IEnumerable<FormatViewDefinition> ViewsOf_Microsoft_PowerShell_Commands_TestConnectionCommand_PingMtuStatus()
+        {
+            yield return new FormatViewDefinition(
+                "Microsoft.PowerShell.Commands.TestConnectionCommand+PingMtuStatus",
+                TableControl.Create()
+                    .AddHeader(Alignment.Left, label: "Source", width: 16)
+                    .AddHeader(Alignment.Left, label: "Address", width: 25)
+                    .AddHeader(Alignment.Right, label: "Latency(ms)", width: 7)
+                    .AddHeader(Alignment.Left, label: "Status", width: 16)
+                    .AddHeader(Alignment.Right, label: "MtuSize(B)", width: 7)
+                    .StartRowDefinition()
+                        .AddPropertyColumn("Source")
+                        .AddScriptBlockColumn(@"
+                            if ($_.Address) {
+                                $_.Address
+                            }
+                            else {
+                                '*'
+                            }
+                        ")
+                        .AddScriptBlockColumn(@"
+                            if ($_.Status -eq 'TimedOut') {
+                                '*'
+                            }
+                            else {
+                                $_.Latency
+                            }
+                        ")
+                        .AddPropertyColumn("Status")
+                        .AddPropertyColumn("MtuSize")
+                    .EndRowDefinition()
+                    .GroupByProperty("Destination")
+                .EndTable());
+        }
+
+        private static IEnumerable<FormatViewDefinition> ViewsOf_Microsoft_PowerShell_Commands_TestConnectionCommand_TraceStatus()
+        {
+            yield return new FormatViewDefinition(
+                "Microsoft.PowerShell.Commands.TestConnectionCommand+TraceStatus",
```

Comment:
I would definitely like to be. This isn't possible that I can see with the standard formatting options, though, at least not without creating a much larger table and also delaying output until all 3 intermediate results are received, since the table formatter doesn't support partial drawing of a result line, nor grouping objects into a single line.

Our options (other than current behaviour) basically are:

1. Create a custom formatter to handle this (I do not know how to do this and would need some pointers)
2. Create a secondary cmdlet or function purely for the display which calls Test-Connection and consumes this output to create display similar to classic ping utils (name suggestion: `Show-Connection`?)
3. Group output by a combination of the target hostname and the hop # (so the same as now, but with each intermediate hop rendered into a separate table)
4. Have TraceStatus contain three PingStatus objects rather than one; Latency becomes a property that contains all three latency values. This works OK, but we would need a `DisplayLatency` property to help the formatter render the three latency values in a readable way. It would also force us to delay output of results from a hop until all three pings are returned (up to 3-4 seconds if all results are TimeOut with the default timeout values).

I think that the most effective options of the above would be 2 or 4. 1 is a significant amount of work, I don't know how good the support for custom formatters is. Option 4 is something I had experimented with previously but ran into a few issues that led me to try the current approach.

Reply:
I suggest to address this in follow PR.